### PR TITLE
Disable TM when generating portable code

### PR DESCRIPTION
**Pull request by dsouzai**
This PR disables the TM feature when generating portable code. 

Fixes https://github.com/eclipse-openj9/openj9/issues/16009

### CODE REVIEW

File: runtime/compiler/env/J9CompilerEnv.cpp [hunk 1]
```diff
@@ -46,8 +46,16 @@ J9::CompilerEnv::initializeTargetEnvironment()
    OMR::CompilerEnvConnector::initializeTargetEnvironment();
    }
 
+void
+J9::CompilerEnv::initializeRelocatableTargetEnvironment()
+   {
+   OMR::CompilerEnvConnector::initializeRelocatableTargetEnvironment();
+   if (J9_ARE_ANY_BITS_SET(javaVM->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_PORTABLE_SHARED_CACHE))
```
Comment (dsouzai) [inline runtime/compiler/env/J9CompilerEnv.cpp:53]: I verified that at the time this is called, the `J9_EXTENDED_RUNTIME2_ENABLE_PORTABLE_SHARED_CACHE` does get set.

File: runtime/compiler/runtime/RelocationRuntime.cpp [hunk 2]
```diff
@@ -1057,7 +1057,7 @@ TR_SharedCacheRelocationRuntime::checkAOTHeaderFlags(const TR_AOTHeader *hdrInCa
    {
    bool defaultMessage = true;
 
-   if (!TR::Compiler->target.cpu.isCompatible(hdrInCache->processorDescription))
+   if (!TR::Compiler->relocatableTarget.cpu.isCompatible(hdrInCache->processorDescription))
```
Comment (dsouzai) [inline runtime/compiler/runtime/RelocationRuntime.cpp:1060]: Strictly speaking, the `target` should be a superset of the `relocatableTarget`, so even if the relocatable target of the code in the SCC wasn't compatible with the relocatable target of the running JVM, but it **is** compatible with the target of the running JVM, it is ok to run the code. However, we probably don't want to allow such a scenario, so I figure it's probably best to use the relocatable target.

**Comment by dsouzai** [comment]
@mpirvu could you please review.

**Review by mpirvu — APPROVED** [review]
LGTM

File: runtime/compiler/z/env/J9CPU.cpp [hunk 3]
```diff
@@ -52,11 +52,17 @@ J9::Z::CPU::detectRelocatable(OMRPortLibrary * const omrPortLib)
...
-      { 
+      {
       processorDescription.processor = OMR_PROCESSOR_S390_Z10;
       processorDescription.physicalProcessor = OMR_PROCESSOR_S390_Z10;
       }
 
+   const uint32_t disabledFeatures [] = { OMR_FEATURE_S390_TE };
+   for (size_t i = 0; i < sizeof(disabledFeatures)/sizeof(uint32_t); i++)
+      {
+      omrsysinfo_processor_set_feature(&portableProcessorDescription, disabledFeatures[i], FALSE);
```
Comment (dsouzai) [inline runtime/compiler/z/env/J9CPU.cpp:63]: Just realized this won't build -_-

Comment (dsouzai) [reply runtime/compiler/z/env/J9CPU.cpp]: Updated, see https://github.com/eclipse-openj9/openj9/compare/9cc24666b8d48ae934ff9a9f822e72c2ed6864b1..b86eb7717a858a50437edbf29364ed3b0d9c9d26

**Comment by dsouzai** [comment]
jenkins test sanity plinux,xlinux,zlinux jdk17

**Comment by dsouzai** [comment]
@mpirvu should be good to merge

**Pull request merged**